### PR TITLE
[Issue-157] Fix styling of icons in the <Form> component

### PR DESCRIPTION
**Pull request by ahmed-gz**
## Description

[Github issue](https://github.com/slashid/javascript/issues/157)

Fix styling of icons in the `<Form>` component following Significa's last feedback: 
[On the authenticating state of the sign-in/sign-up component, there are some assets missing.pdf](https://github.com/slashid/javascript/files/12601184/On.the.authenticating.state.of.the.sign-in.sign-up.25a57aff4bd04bcb9ec48c6e577ad98b.pdf)

### Status: `authenticating` - method: `sms_link`
image

### Status: `authenticating` - method: `email_link`
image

### Status: `authenticating` - method: others
image

### Status: **error**
image

### Status: **success**
image

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have generated the new version of the docs website and smoke tested it
- [ ] I have checked that my changes haven't caused semantic errors in the existing docs
- [x] I have generated a `changeset` if my change affects the published packages

### CODE REVIEW

**Review by jake-slashid — APPROVED** [review]

**Comment by ahmed-gz** [comment]
> gradients



> gradients

As far as I know the spinner icon is the only thing we are supposed to animate which was already handled previously, Significa didn't mention any missing animation in their feedback but I will double check with Ines and let you know, I will mark this PR as draft until I have an answer.

**Comment by ahmed-gz** [comment]
> > gradients
> 
> > gradients
> 
> As far as I know the spinner icon is the only thing we are supposed to animate which was already handled previously, Significa didn't mention any missing animation in their feedback but I will double check with Ines and let you know, I will mark this PR as draft until I have an answer.

@jake-slashid Until I get a confirmation from Ines about the animation, you can check how the icons actually look in storybook https://157-form-icon.website-react-storybook.pages.dev/?path=/docs/components-dynamic-flow--docs&globals=backgrounds.value:!hex(F8F8F8)

**Comment by ahmed-gz** [comment]
@jake-slashid You were right about the gradient animation, I got a confirmation about that from Ines:
image

I found the animation in the design flow [here](https://www.figma.com/proto/nOXz0jhwoHGGXJwOdfTmsa/SlashID-%E2%80%94-Sign-in%2FSign-up-Design-by-Significa?type=design&node-id=204-3591&t=OHIbIuUAxb8uJBFQ-0&scaling=min-zoom&page-id=204%3A3338&starting-point-node-id=204%3A3585&show-proto-sidebar=1), I will try to get as close to it as possible because the design doesn't have the animation CSS code sadly.

**Comment by jake-slashid** [comment]
It was just a hunch! It felt a bit bad having the loading icon without motion 😅 

The animation looks nice

**Comment by ahmed-gz** [comment]
@jake-slashid It took me a while to build the animation as it was quite complex to sync all the gradient pieces, please check it out and let me know how it looks!

**Comment by jake-slashid** [comment]
It's beautiful! 🤩

**Comment by jake-slashid** [comment]
My only feedback is that compared to the figma prototype it feels a little slower to start, and generally a little slower moving while animating

**Review by jake-slashid — APPROVED** [review]

**Comment by ahmed-gz** [comment]
> figma

Thanks a lot for you patience on this, I can never get the timing right when I look at an animation in figma without feedback 😅 I wish figma would include animation code similar to layout and other stuff in the future to make it easy to match the design accurately.
I will make it a bit faster and let you know so you can test it.

**Comment by ahmed-gz** [comment]
@jake-slashid I made the animation faster, could you check how it looks now?

**Comment by jake-slashid** [comment]
Looks great and my gut feel says that it feels better too, much more responsive/snappy

**Comment by jake-slashid** [comment]
Is it right for the error to animate? There is no "pending" nature to this state, so intuitively it feels wrong.

5ec066ccc536247aa3866e17eada1a7f

**Comment by jake-slashid** [comment]
https://github.com/slashid/javascript/assets/139138253/39a241b3-0b83-4f12-ada4-ec769e701a2c

**Comment by ahmed-gz** [comment]
> Is it right for the error to animate? There is no "pending" nature to this state, so intuitively to me it feels wrong.

Nice catch, you're totally right, I will remove animation for that state.

**Comment by ahmed-gz** [comment]
> Is it right for the error to animate? There is no "pending" nature to this state, so intuitively to me it feels wrong.

Error state don't animate now, let me know if it looks good.

**Comment by jake-slashid** [comment]
Looks good to me 👌

**Review by ikovic — APPROVED** [review]

File: packages/react/src/domain/handles.ts [hunk 1]
```diff
@@ -52,6 +58,14 @@ export function isFactorOidc(factor: Factor): factor is FactorOIDC {
   return factor.method === "oidc";
 }
 
+export function isFactorEmailLink(factor: Factor): boolean {
+  return factor.method === "email_link";
+}
+
+export function isFactorSmsLink(factor: Factor): boolean {
```
Comment (ikovic) [inline packages/react/src/domain/handles.ts:65]: These would be a bit more useful in the form of type guards - can you check the existing `is*` functions in this module for inspiration?

**Pull request merged**